index.js

```diff
@@ -74,9 +75,16 @@ export default class LinterJSCS {
     require('atom-package-deps').install();
 
     this.observer = atom.workspace.observeTextEditors((editor) => {
-      editor.getBuffer().onDidSave(() => {
-        if (grammarScopes.indexOf(editor.getGrammar().scopeName) !== -1 && this.fixOnSave) {
-          this.fixString(editor);
+      editor.getBuffer().onWillSave(() => {
+
+        // Exclude `excludeFiles` for fix on save
+        const config = this.getConfig(editor.getPath());
+        var exclude = globule.isMatch(config && config.excludeFiles, this.getFilePath(editor.getPath()));
+
+        if (grammarScopes.indexOf(editor.getGrammar().scopeName) !== -1 && this.fixOnSave && !exclude) {
+          process.nextTick(() => {
+            this.fixString();
```

Comment:
Explain more about the race condition, I don't understand. 

And that's a great question about `nextTick` not sure why its like that.